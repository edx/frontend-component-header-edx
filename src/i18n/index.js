import arMessages from './messages/ar.json';
// no need to import en messages-- they are in the defaultMessage field

import caMessages from './messages/ca.json';
import heMessages from './messages/he.json';
import idMessages from './messages/id.json';
import plMessages from './messages/pl.json';
import ruMessages from './messages/ru.json';
import thMessages from './messages/th.json';
import ukMessages from './messages/uk.json';

import es419Messages from './messages/es_419.json';
import frMessages from './messages/fr.json';
import kokrMessages from './messages/ko_KR.json';
import ptbrMessages from './messages/pt_BR.json';
import zhcnMessages from './messages/zh_CN.json';

const messages = {
  ar: arMessages,
  ca: caMessages,
  he: heMessages,
  id: idMessages,
  pl: plMessages,
  ru: ruMessages,
  th: thMessages,
  uk: ukMessages,
  'es-419': es419Messages,
  fr: frMessages,
  'zh-cn': zhcnMessages,
  'ko-kr': kokrMessages,
  'pt-br': ptbrMessages,
};

export default messages;
