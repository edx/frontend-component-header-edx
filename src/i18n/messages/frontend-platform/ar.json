{
  "unexpected.error.message.text": "حدث خطأ غير متوقع. الرجاء الضغط على الزر أدناه لتحديث الصفحة.",
  "unexpected.error.button.text": "حاول ثانية"
}