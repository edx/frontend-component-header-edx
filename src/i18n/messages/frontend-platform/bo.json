{
  "unexpected.error.message.text": "ངེས་མེད་ཀྱི་གནད་དོན་ཞིག་བྱུང་སོང་བས། གཤམ་གསལ་གྱི་ཐེབས་དེར་བསྣུན་ནས་ཤོག་ངོས་དེ་བསྐྱར་སྐོར་ཞིག་བྱོས།",
  "unexpected.error.button.text": "བསྐྱར་དུ་ཚོད་ལྟ་བྱོས།"
}