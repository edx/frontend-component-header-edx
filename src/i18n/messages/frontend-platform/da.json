{
  "unexpected.error.message.text": "Der opstod en uventet fejl. Klik venligst på knappen nedenfor for at opdatere siden.",
  "unexpected.error.button.text": "Prøv igen"
}