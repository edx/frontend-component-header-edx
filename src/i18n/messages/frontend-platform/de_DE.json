{
  "unexpected.error.message.text": "Ein unerwarteter Fehler ist aufgetreten. Bitte klicken Sie auf die SchaltflÃ¤che unten, um die Seite zu aktualisieren.",
  "unexpected.error.button.text": "Erneut versuchen"
}