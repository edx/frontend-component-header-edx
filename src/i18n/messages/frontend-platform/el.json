{
  "unexpected.error.message.text": "Προέκυψε ένα μη αναμενόμενο σφάλμα. Κάντε κλικ στο κουμπί παρακάτω για να ανανεώσετε τη σελίδα.",
  "unexpected.error.button.text": "Προσπάθησε ξανά"
}