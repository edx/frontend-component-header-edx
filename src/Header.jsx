import React, { useContext } from 'react';
import Responsive from 'react-responsive';
import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
import { sendTrackEvent } from '@edx/frontend-platform/analytics';

import { AppContext } from '@edx/frontend-platform/react';
import {
  APP_CONFIG_INITIALIZED,
  ensureConfig,
  mergeConfig,
  getConfig,
  subscribe,
} from '@edx/frontend-platform';

import DesktopHeader from './DesktopHeader';
import MobileHeader from './MobileHeader';

import LogoSVG from './logo.svg';

import messages from './Header.messages';

import useEnterpriseConfig from './data/hooks/enterprise';

ensureConfig([
  'LMS_BASE_URL',
  'LOGOUT_URL',
  'LOGIN_URL',
  'MARKETING_SITE_BASE_URL',
  'ORDER_HISTORY_URL',
], 'Header component');

subscribe(APP_CONFIG_INITIALIZED, () => {
  mergeConfig({
    MINIMAL_HEADER: !!process.env.MINIMAL_HEADER,
    ENTERPRISE_LEARNER_PORTAL_HOSTNAME: process.env.ENTERPRISE_LEARNER_PORTAL_HOSTNAME,
  }, 'Header additional config');
});

function Header({ intl }) {
  const { authenticatedUser, config } = useContext(AppContext);
  const {
    enterpriseLearnerPortalLink,
    enterpriseCustomerBrandingConfig,
  } = useEnterpriseConfig(authenticatedUser, config);

  const mainMenu = [
    {
      type: 'item',
      href: `${config.LMS_BASE_URL}/dashboard`,
      content: intl.formatMessage(messages['header.links.courses']),
    },
    {
      type: 'item',
      href: `${config.LMS_BASE_URL}/dashboard/programs`,
      content: intl.formatMessage(messages['header.links.programs']),
    },
    {
      type: 'item',
      href: `${config.MARKETING_SITE_BASE_URL}/course`,
      content: intl.formatMessage(messages['header.links.content.search']),
      onClick: () => {
        sendTrackEvent(
          'edx.bi.dashboard.find_courses_button.clicked',
          { category: 'header', label: 'header' },
        );
      },
    },
  ];

  const dashboardMenuItem = {
    type: 'item',
    href: `${config.LMS_BASE_URL}/dashboard`,
    content: intl.formatMessage(messages['header.user.menu.dashboard']),
  };

  const logoutMenuItem = {
    type: 'item',
    href: config.LOGOUT_URL,
    content: intl.formatMessage(messages['header.user.menu.logout']),
  };

  const orderHistoryItem = {
    type: 'item',
    href: config.ORDER_HISTORY_URL,
    content: intl.formatMessage(messages['header.user.menu.order.history']),
  };

  // If there is an Enterprise LP link, use that instead of the B2C Dashboard
  let baseUserMenuDashboardLinks = [];
  if (enterpriseLearnerPortalLink) {
    baseUserMenuDashboardLinks = [enterpriseLearnerPortalLink];
  } else {
    baseUserMenuDashboardLinks = [dashboardMenuItem];
  }

  let userMenu = authenticatedUser === null ? [] : [
    ...baseUserMenuDashboardLinks,
    {
      type: 'item',
      href: `${config.LMS_BASE_URL}/u/${authenticatedUser.username}`,
      content: intl.formatMessage(messages['header.user.menu.profile']),
    },
    {
      type: 'item',
      href: `${config.LMS_BASE_URL}/account/settings`,
      content: intl.formatMessage(messages['header.user.menu.account.settings']),
    },
    logoutMenuItem,
  ];

  // Users should only see Order History if they do not have an available
  // learner portal, because an available learner portal currently means
  // that they access content via Subscriptions, in which context an "order"
  // is not relevant.
  if (!enterpriseLearnerPortalLink) {
    userMenu.splice(-1, 0, orderHistoryItem);
  }

  if (getConfig().MINIMAL_HEADER && authenticatedUser !== null) {
    userMenu = [
      dashboardMenuItem,
      logoutMenuItem,
    ];
  }

  const loggedOutItems = [
    {
      type: 'item',
      href: config.LOGIN_URL,
      content: intl.formatMessage(messages['header.user.menu.login']),
    },
    {
      type: 'item',
      href: `${config.LMS_BASE_URL}/register`,
      content: intl.formatMessage(messages['header.user.menu.register']),
    },
  ];

  let props = {
    logo: LogoSVG,
    logoAltText: 'edX',
    siteName: 'edX',
    logoDestination: getConfig().MINIMAL_HEADER ? null : `${config.LMS_BASE_URL}/dashboard`,
    loggedIn: authenticatedUser !== null,
    username: authenticatedUser !== null ? authenticatedUser.username : null,
    avatar: authenticatedUser !== null ? authenticatedUser.avatar : null,
    mainMenu: getConfig().MINIMAL_HEADER ? [] : mainMenu,
    userMenu,
    loggedOutItems,
  };

  if (enterpriseCustomerBrandingConfig) {
    props = {
      ...props,
      ...enterpriseCustomerBrandingConfig,
    };
  }

  return (
    <React.Fragment>
      <Responsive maxWidth={768}>
        <MobileHeader {...props} />
      </Responsive>
      <Responsive minWidth={769}>
        <DesktopHeader {...props} />
      </Responsive>
    </React.Fragment>
  );
}

Header.propTypes = {
  intl: intlShape.isRequired,
};

export default injectIntl(Header);
