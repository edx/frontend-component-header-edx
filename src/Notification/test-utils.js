import MockAdapter from 'axios-mock-adapter';
import { Factory } from 'rosie';

import { getAuthenticatedHttpClient } from '@edx/frontend-platform/auth';

import {
  getNotificationsListApiUrl, getNotificationsCountApiUrl, markNotificationsSeenApiUrl, markNotificationAsReadApiUrl,
} from './data/api';

import './data/__factories__';

const notificationCountsApiUrl = getNotificationsCountApiUrl();
const notificationsApiUrl = getNotificationsListApiUrl();
const markedAllNotificationsAsSeenApiUrl = markNotificationsSeenApiUrl('discussion');
const markedAllNotificationsAsReadApiUrl = markNotificationAsReadApiUrl();

export default async function mockNotificationsResponse(todaycount = 8, earlierCount = 2) {
  const axiosMock = new MockAdapter(getAuthenticatedHttpClient());
  const notifications = (Factory.buildList('notification', todaycount, null, { createdDate: new Date().toISOString() }).concat(
    Factory.buildList('notification', earlierCount, null, { createdDate: '2023-06-01T00:46:11.979531Z' }),
  ));
  axiosMock.onGet(notificationCountsApiUrl).reply(200, (Factory.build('notificationsCount')));
  axiosMock.onPut(markedAllNotificationsAsSeenApiUrl).reply(200, { message: 'Notifications marked seen.' });
  axiosMock.onPatch(markedAllNotificationsAsReadApiUrl).reply(200, { message: 'Notifications marked read.' });

  axiosMock.onGet(notificationsApiUrl).reply(200, (Factory.build('notificationsList', {
    results: notifications,
    num_pages: 2,
    current_page: 2,
    next: `${notificationsApiUrl}?app_name=discussion&page=2`,
  })));
}
