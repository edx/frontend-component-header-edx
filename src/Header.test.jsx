
import React from 'react';
import { act } from 'react-dom/test-utils';
import { IntlProvider } from '@edx/frontend-platform/i18n';
import TestRenderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { getLearnerPortalLinks } from '@edx/frontend-enterprise';
import { AppContext } from '@edx/frontend-platform/react';
import { getConfig } from '@edx/frontend-platform';
import { Context as ResponsiveContext } from 'react-responsive';

import Header from './index';

jest.mock('@edx/frontend-platform');
jest.mock('@edx/frontend-enterprise');

getConfig.mockReturnValue({});

describe('<Header />', () => {
  beforeEach(() => {
    getLearnerPortalLinks.mockReturnValue(Promise.resolve([]));
  });

  const mockLearnerPortalLinks = () => {
    getLearnerPortalLinks.mockReturnValue(Promise.resolve([
      {
        url: 'http://localhost:8000',
        title: 'My Enterprise',
        branding_configuration: {
          logo: 'my-logo',
        },
      },
    ]));
  };

  it('renders correctly for unauthenticated users on desktop', () => {
    const component = (
      <ResponsiveContext.Provider value={{ width: 1280 }}>
        <IntlProvider locale="en" messages={{}}>
          <AppContext.Provider value={{
            authenticatedUser: null,
            config: {
              LMS_BASE_URL: process.env.LMS_BASE_URL,
              LOGIN_URL: process.env.LOGIN_URL,
              LOGOUT_URL: process.env.LOGOUT_URL,
              MARKETING_SITE_BASE_URL: process.env.MARKETING_SITE_BASE_URL,
              ORDER_HISTORY_URL: process.env.ORDER_HISTORY_URL,
            },
            }}
          >
            <Header />
          </AppContext.Provider>
        </IntlProvider>
      </ResponsiveContext.Provider>
    );

    const wrapper = TestRenderer.create(component);

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('renders correctly for authenticated users on desktop', () => {
    const component = (
      <ResponsiveContext.Provider value={{ width: 1280 }}>
        <IntlProvider locale="en" messages={{}}>
          <AppContext.Provider value={{
            authenticatedUser: {
              userId: 'abc123',
              username: 'edX',
              roles: [],
              administrator: false,
            },
            config: {
              LMS_BASE_URL: process.env.LMS_BASE_URL,
              LOGIN_URL: process.env.LOGIN_URL,
              LOGOUT_URL: process.env.LOGOUT_URL,
              MARKETING_SITE_BASE_URL: process.env.MARKETING_SITE_BASE_URL,
              ORDER_HISTORY_URL: process.env.ORDER_HISTORY_URL,
            },
            }}
          >
            <Header />
          </AppContext.Provider>
        </IntlProvider>
      </ResponsiveContext.Provider>
    );

    const wrapper = TestRenderer.create(component);

    expect(wrapper.toJSON()).toMatchSnapshot();
  });


  it('renders correctly for authenticated users on desktop with or without learner portal links', async () => {
    const component = (
      <ResponsiveContext.Provider value={{ width: 1280 }}>
        <IntlProvider locale="en" messages={{}}>
          <AppContext.Provider value={{
            authenticatedUser: {
              userId: 'abc123',
              username: 'edX',
              roles: [],
              administrator: false,
            },
            config: {
              LMS_BASE_URL: process.env.LMS_BASE_URL,
              LOGIN_URL: process.env.LOGIN_URL,
              LOGOUT_URL: process.env.LOGOUT_URL,
              MARKETING_SITE_BASE_URL: process.env.MARKETING_SITE_BASE_URL,
              ORDER_HISTORY_URL: process.env.ORDER_HISTORY_URL,
            },
            }}
          >
            <Header />
          </AppContext.Provider>
        </IntlProvider>
      </ResponsiveContext.Provider>
    );

    // When learner portal links are not present, Order History should be a dropdown item
    let wrapper = mount(component);
    const flushPromises = () => new Promise(setImmediate);
    await act(async () => {
      await flushPromises();
    });
    wrapper.update();
    wrapper.find('.menu-trigger').simulate('click');
    expect(wrapper.find('a[children="Order History"]')).toHaveLength(1);
    expect(wrapper.find('a[children="Dashboard"]')).toHaveLength(1);

    // When learner portal links are present, Order History should not be a dropdown item
    // We do this in the same test to avoid weird things about jest wanting you to use
    // the "correct" act() function (even though it gives the same warning regardless
    // of where you import it from, be it react-dom/test-utils or react-test-renderer).
    mockLearnerPortalLinks();
    wrapper = mount(component);
    await act(async () => {
      await flushPromises();
    });
    wrapper.update();
    wrapper.find('.menu-trigger').simulate('click');
    expect(wrapper.find('a[children="Order History"]')).toHaveLength(0);
    expect(wrapper.find('a[children="Dashboard"]')).toHaveLength(1);
  });

  it('renders correctly for unauthenticated users on mobile', () => {
    const component = (
      <ResponsiveContext.Provider value={{ width: 500 }}>
        <IntlProvider locale="en" messages={{}}>
          <AppContext.Provider value={{
            authenticatedUser: null,
            config: {
              LMS_BASE_URL: process.env.LMS_BASE_URL,
              LOGIN_URL: process.env.LOGIN_URL,
              LOGOUT_URL: process.env.LOGOUT_URL,
              MARKETING_SITE_BASE_URL: process.env.MARKETING_SITE_BASE_URL,
              ORDER_HISTORY_URL: process.env.ORDER_HISTORY_URL,
            },
            }}
          >
            <Header />
          </AppContext.Provider>
        </IntlProvider>
      </ResponsiveContext.Provider>
    );

    const wrapper = TestRenderer.create(component);

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('renders correctly for authenticated users on mobile', () => {
    const component = (
      <ResponsiveContext.Provider value={{ width: 500 }}>
        <IntlProvider locale="en" messages={{}}>
          <AppContext.Provider value={{
            authenticatedUser: {
              userId: 'abc123',
              username: 'edX',
              roles: [],
              administrator: false,
            },
            config: {
              LMS_BASE_URL: process.env.LMS_BASE_URL,
              LOGIN_URL: process.env.LOGIN_URL,
              LOGOUT_URL: process.env.LOGOUT_URL,
              MARKETING_SITE_BASE_URL: process.env.MARKETING_SITE_BASE_URL,
              ORDER_HISTORY_URL: process.env.ORDER_HISTORY_URL,
            },
            }}
          >
            <Header />
          </AppContext.Provider>
        </IntlProvider>
      </ResponsiveContext.Provider>
    );

    const wrapper = TestRenderer.create(component);

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  describe('minimal', () => {
    beforeEach(() => {
      getConfig.mockReturnValue({
        MINIMAL_HEADER: true,
      });
    });

    afterEach(() => {
      getConfig.mockReturnValue({
        MINIMAL_HEADER: false,
      });
    });

    it('renders correctly for unauthenticated users when minimal', () => {
      const component = (
        <ResponsiveContext.Provider value={{ width: 1280 }}>
          <IntlProvider locale="en" messages={{}}>
            <AppContext.Provider value={{
              authenticatedUser: null,
              config: {
                LMS_BASE_URL: process.env.LMS_BASE_URL,
                LOGIN_URL: process.env.LOGIN_URL,
                LOGOUT_URL: process.env.LOGOUT_URL,
                MARKETING_SITE_BASE_URL: process.env.MARKETING_SITE_BASE_URL,
                ORDER_HISTORY_URL: process.env.ORDER_HISTORY_URL,
              },
              }}
            >
              <Header />
            </AppContext.Provider>
          </IntlProvider>
        </ResponsiveContext.Provider>
      );

      const wrapper = TestRenderer.create(component);

      expect(wrapper.toJSON()).toMatchSnapshot();
    });

    it('renders correctly for authenticated users when minimal', () => {
      const component = (
        <ResponsiveContext.Provider value={{ width: 1280 }}>
          <IntlProvider locale="en" messages={{}}>
            <AppContext.Provider value={{
              authenticatedUser: {
                userId: 'abc123',
                username: 'edX',
                roles: [],
                administrator: false,
              },
              config: {
                LMS_BASE_URL: process.env.LMS_BASE_URL,
                LOGIN_URL: process.env.LOGIN_URL,
                LOGOUT_URL: process.env.LOGOUT_URL,
                MARKETING_SITE_BASE_URL: process.env.MARKETING_SITE_BASE_URL,
                ORDER_HISTORY_URL: process.env.ORDER_HISTORY_URL,
              },
              }}
            >
              <Header />
            </AppContext.Provider>
          </IntlProvider>
        </ResponsiveContext.Provider>
      );

      const wrapper = TestRenderer.create(component);

      expect(wrapper.toJSON()).toMatchSnapshot();
    });
  });
});
