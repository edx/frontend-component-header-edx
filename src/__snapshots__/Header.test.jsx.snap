// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> minimal renders correctly for authenticated users when minimal 1`] = `
<header
  className="site-header-desktop"
>
  <a
    className="nav-skip sr-only sr-only-focusable"
    href="#main"
  >
    Skip to main content
  </a>
  <div
    className="container-fluid null"
  >
    <div
      className="nav-container position-relative d-flex align-items-center"
    >
      <img
        alt="edX"
        className="logo"
        src="https://edx-cdn.org/v3/prod/logo.svg"
      />
      <nav
        aria-label="Main"
        className="nav main-nav"
      />
      <nav
        aria-label="Secondary"
        className="nav secondary-menu-container align-items-center ml-auto"
      >
        <div
          className="dropdown"
          onKeyDown={[Function]}
        >
          <button
            alt=""
            aria-expanded={false}
            aria-haspopup={true}
            aria-label="Account menu for edX"
            className="btn-avatar pgn__avatar-button-avatar pgn__avatar-button-avatar-md dropdown-toggle btn btn-tertiary btn-md"
            data-hj-suppress={true}
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              className="pgn__avatar-button-avatar-wrap"
            >
              <img
                alt=""
                className="pgn__avatar pgn__avatar-sm"
                src="icon/mock/path"
              />
            </span>
            edX
          </button>
        </div>
      </nav>
    </div>
  </div>
</header>
`;

exports[`<Header /> minimal renders correctly for unauthenticated users when minimal 1`] = `
<header
  className="site-header-desktop"
>
  <a
    className="nav-skip sr-only sr-only-focusable"
    href="#main"
  >
    Skip to main content
  </a>
  <div
    className="container-fluid null"
  >
    <div
      className="nav-container position-relative d-flex align-items-center"
    >
      <img
        alt="edX"
        className="logo"
        src="https://edx-cdn.org/v3/prod/logo.svg"
      />
      <nav
        aria-label="Main"
        className="nav main-nav"
      />
      <nav
        aria-label="Secondary"
        className="nav secondary-menu-container align-items-center ml-auto"
      >
        <a
          className="btn mr-2 btn-link"
          href="http://localhost:18000/login"
        >
          Login
        </a>
        <a
          className="btn mr-2 btn-outline-primary"
          href="http://localhost:18000/register"
        >
          Sign Up
        </a>
      </nav>
    </div>
  </div>
</header>
`;

exports[`<Header /> renders correctly for authenticated users on desktop 1`] = `
<header
  className="site-header-desktop"
>
  <a
    className="nav-skip sr-only sr-only-focusable"
    href="#main"
  >
    Skip to main content
  </a>
  <div
    className="container-fluid null"
  >
    <div
      className="nav-container position-relative d-flex align-items-center"
    >
      <a
        className="logo"
        href="http://localhost:18000/dashboard"
      >
        <img
          alt="edX"
          className="d-block"
          src="https://edx-cdn.org/v3/prod/logo.svg"
        />
      </a>
      <nav
        aria-label="Main"
        className="nav main-nav"
      >
        <a
          className="nav-link"
          href="http://localhost:18000/dashboard"
        >
          Courses
        </a>
        <a
          className="nav-link"
          href="http://localhost:18000/dashboard/programs"
        >
          Programs
        </a>
        <a
          className="nav-link"
          href="http://localhost:18000/course"
        >
          Discover New
        </a>
      </nav>
      <nav
        aria-label="Secondary"
        className="nav secondary-menu-container align-items-center ml-auto"
      >
        <div
          className="dropdown"
          onKeyDown={[Function]}
        >
          <button
            alt=""
            aria-expanded={false}
            aria-haspopup={true}
            aria-label="Account menu for edX"
            className="btn-avatar pgn__avatar-button-avatar pgn__avatar-button-avatar-md dropdown-toggle btn btn-tertiary btn-md"
            data-hj-suppress={true}
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              className="pgn__avatar-button-avatar-wrap"
            >
              <img
                alt=""
                className="pgn__avatar pgn__avatar-sm"
                src="icon/mock/path"
              />
            </span>
            edX
          </button>
        </div>
      </nav>
    </div>
  </div>
</header>
`;

exports[`<Header /> renders correctly for authenticated users on mobile 1`] = `
<header
  aria-label="Main"
  className="site-header-mobile d-flex justify-content-between align-items-center shadow sticky-top"
>
  <a
    className="nav-skip sr-only sr-only-focusable"
    href="#main"
  >
    Skip to main content
  </a>
  <div
    className="w-100 d-flex justify-content-start"
  >
    <div
      className="menu position-static"
      onKeyDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <button
        aria-expanded={false}
        aria-haspopup="menu"
        aria-label="Main Menu"
        className="menu-trigger icon-button"
        onClick={[Function]}
        title="Main Menu"
      >
        <svg
          aria-hidden={true}
          focusable="false"
          height="24px"
          role="img"
          style={
            Object {
              "height": "1.5rem",
              "width": "1.5rem",
            }
          }
          version="1.1"
          viewBox="0 0 24 24"
          width="24px"
        >
          <rect
            fill="currentColor"
            height="2"
            width="20"
            x="2"
            y="5"
          />
          <rect
            fill="currentColor"
            height="2"
            width="20"
            x="2"
            y="11"
          />
          <rect
            fill="currentColor"
            height="2"
            width="20"
            x="2"
            y="17"
          />
        </svg>
      </button>
    </div>
  </div>
  <div
    className="w-100 d-flex justify-content-center"
  >
    <a
      className="logo"
      href="http://localhost:18000/dashboard"
      itemType="http://schema.org/Organization"
    >
      <img
        alt="edX"
        className="d-block"
        src="https://edx-cdn.org/v3/prod/logo.svg"
      />
    </a>
  </div>
  <div
    className="w-100 d-flex justify-content-end align-items-center"
  >
    <nav
      aria-label="Secondary"
      className="menu position-static"
      onKeyDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <button
        aria-expanded={false}
        aria-haspopup="menu"
        aria-label="Account Menu"
        className="btn-avatar pgn__avatar-button-avatar pgn__avatar-button-avatar-md menu-trigger null pgn__avatar-button-hide-label btn btn-tertiary btn-md"
        disabled={false}
        onClick={[Function]}
        title="Account Menu"
        type="button"
      >
        <span
          className="pgn__avatar-button-avatar-wrap"
        >
          <img
            alt="edX"
            className="pgn__avatar pgn__avatar-sm"
            src="icon/mock/path"
          />
        </span>
      </button>
    </nav>
  </div>
</header>
`;

exports[`<Header /> renders correctly for unauthenticated users on desktop 1`] = `
<header
  className="site-header-desktop"
>
  <a
    className="nav-skip sr-only sr-only-focusable"
    href="#main"
  >
    Skip to main content
  </a>
  <div
    className="container-fluid null"
  >
    <div
      className="nav-container position-relative d-flex align-items-center"
    >
      <a
        className="logo"
        href="http://localhost:18000/dashboard"
      >
        <img
          alt="edX"
          className="d-block"
          src="https://edx-cdn.org/v3/prod/logo.svg"
        />
      </a>
      <nav
        aria-label="Main"
        className="nav main-nav"
      >
        <a
          className="nav-link"
          href="http://localhost:18000/dashboard"
        >
          Courses
        </a>
        <a
          className="nav-link"
          href="http://localhost:18000/dashboard/programs"
        >
          Programs
        </a>
        <a
          className="nav-link"
          href="http://localhost:18000/course"
        >
          Discover New
        </a>
      </nav>
      <nav
        aria-label="Secondary"
        className="nav secondary-menu-container align-items-center ml-auto"
      >
        <a
          className="btn mr-2 btn-link"
          href="http://localhost:18000/login"
        >
          Login
        </a>
        <a
          className="btn mr-2 btn-outline-primary"
          href="http://localhost:18000/register"
        >
          Sign Up
        </a>
      </nav>
    </div>
  </div>
</header>
`;

exports[`<Header /> renders correctly for unauthenticated users on mobile 1`] = `
<header
  aria-label="Main"
  className="site-header-mobile d-flex justify-content-between align-items-center shadow sticky-top"
>
  <a
    className="nav-skip sr-only sr-only-focusable"
    href="#main"
  >
    Skip to main content
  </a>
  <div
    className="w-100 d-flex justify-content-start"
  >
    <div
      className="menu position-static"
      onKeyDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <button
        aria-expanded={false}
        aria-haspopup="menu"
        aria-label="Main Menu"
        className="menu-trigger icon-button"
        onClick={[Function]}
        title="Main Menu"
      >
        <svg
          aria-hidden={true}
          focusable="false"
          height="24px"
          role="img"
          style={
            Object {
              "height": "1.5rem",
              "width": "1.5rem",
            }
          }
          version="1.1"
          viewBox="0 0 24 24"
          width="24px"
        >
          <rect
            fill="currentColor"
            height="2"
            width="20"
            x="2"
            y="5"
          />
          <rect
            fill="currentColor"
            height="2"
            width="20"
            x="2"
            y="11"
          />
          <rect
            fill="currentColor"
            height="2"
            width="20"
            x="2"
            y="17"
          />
        </svg>
      </button>
    </div>
  </div>
  <div
    className="w-100 d-flex justify-content-center"
  >
    <a
      className="logo"
      href="http://localhost:18000/dashboard"
      itemType="http://schema.org/Organization"
    >
      <img
        alt="edX"
        className="d-block"
        src="https://edx-cdn.org/v3/prod/logo.svg"
      />
    </a>
  </div>
  <div
    className="w-100 d-flex justify-content-end align-items-center"
  >
    <nav
      aria-label="Secondary"
      className="menu position-static"
      onKeyDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <button
        aria-expanded={false}
        aria-haspopup="menu"
        aria-label="Account Menu"
        className="btn-avatar pgn__avatar-button-avatar pgn__avatar-button-avatar-md menu-trigger null pgn__avatar-button-hide-label btn btn-tertiary btn-md"
        disabled={false}
        onClick={[Function]}
        title="Account Menu"
        type="button"
      >
        <span
          className="pgn__avatar-button-avatar-wrap"
        >
          <img
            alt={null}
            className="pgn__avatar pgn__avatar-sm"
            src="icon/mock/path"
          />
        </span>
      </button>
    </nav>
  </div>
</header>
`;
